### PR TITLE
Make `load` work with properties with `_`

### LINKED ISSUE #3: Envattrs does not work when property name contains an underscore
(Will make a PR later)

attrs handles underscores so that `@attr.s;class A: _a: int` becomes `class A: def __init__(self, a: int)` but envattrs doesn't do that transformation

this means the following wouldn't work
```
@attr.s
class Options:
    _port = attr.ib(converter=int)


options = envattrs.load(Options, 'PREFIX') 
```

### PR DESCRIPTION
**Pull request by y26805**
Fix https://github.com/HENNGE/envattrs/issues/3

### CODE REVIEW

File: src/envattrs/__init__.py [hunk 1]
```diff
@@ -27,15 +27,16 @@ def _iter(cls, prefix, source):
 
     for field in fields(cls):
         sub = field.metadata.get(SubAttrs, None)
-        key = f"{prefix.upper()}_{field.name.upper()}"
+        field_name = field.name.strip("_")
```
Comment (ojii) [inline src/envattrs/__init__.py]: I think this should only [`lstrip`](https://github.com/python-attrs/attrs/blob/ccecbf3ef7b56d53cd8d3a52c72e07a488d825d6/src/attr/_make.py#L2168)

Comment (y26805) [reply src/envattrs/__init__.py]: you are absolutely right

File: tests/test_envattrs.py [hunk 2]
```diff
@@ -23,6 +23,24 @@ class Foo:
...
+def test_load_with_leading_underscore():
+    @attr.s
+    class Foo:
+        _bar = attr.ib()
+
+    instance = envattrs.load(Foo, "FOO", {"FOO_BAR": "baz"})
+    assert instance._bar == "baz"
+
+
+def test_load_with_non_leading_underscore():
```
Comment (ojii) [inline tests/test_envattrs.py]: maybe add one with a trailing underscore just to cover all our bases?

Comment (y26805) [reply tests/test_envattrs.py]: done. All your base are belong to us

**Comment by ojii** [comment]
thank you

**Pull request merged**